pkg/controller/v1beta1/inferenceservice/reconcilers/ingress/ingress_reconciler.go

```diff
@@ -311,6 +320,88 @@ func createIngress(isvc *v1beta1.InferenceService, config *v1beta1.IngressConfig
 		hosts = append(hosts, serviceHost)
 		gateways = append(gateways, config.IngressGateway)
 	}
+	if config.UrlTemplate != "" {
```

Comment:
Can we add the test cases for the generated vs matching rules?

Reply:
You mean extend `ingress_reconciler_test.go` to test createIngress() for path-based routing scenarios?